optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/score/descriptor/ScoreDescriptor.java

```diff
@@ -0,0 +1,221 @@
+import org.optaplanner.core.impl.score.buildin.bendable.BendableScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.bendablebigdecimal.BendableBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.bendablelong.BendableLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardmediumsoft.HardMediumSoftScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardmediumsoftlong.HardMediumSoftLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardsoft.HardSoftScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.hardsoftlong.HardSoftLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.simple.SimpleScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.simplebigdecimal.SimpleBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.simplelong.SimpleLongScoreDefinition;
+import org.optaplanner.core.impl.score.definition.ScoreDefinition;
+
+public class ScoreDescriptor {
+    private final MemberAccessor scoreMemberAccessor;
+    private final ScoreDefinition<?> scoreDefinition;
+
+    private ScoreDescriptor(MemberAccessor scoreMemberAccessor, ScoreDefinition<?> scoreDefinition) {
+        this.scoreMemberAccessor = scoreMemberAccessor;
+        this.scoreDefinition = scoreDefinition;
+    }
+
+    public static ScoreDescriptor buildScoreDescriptor(
+            DescriptorPolicy descriptorPolicy,
+            Member member,
+            Class<?> solutionClass) {
+        MemberAccessor scoreMemberAccessor = buildScoreMemberAccessor(descriptorPolicy, member);
+        Class<? extends Score<?>> scoreType = extractScoreType(scoreMemberAccessor, solutionClass);
+        PlanningScore annotation = scoreMemberAccessor.getAnnotation(PlanningScore.class);
+        if (annotation == null) {
+            // The member was autodiscovered
+            annotation = AutoDiscoverAnnotationDefaults.getPlanningScoreAnnotation();
+        }
+        ScoreDefinition<?> scoreDefinition = buildScoreDefinition(solutionClass, scoreMemberAccessor, scoreType, annotation);
+        return new ScoreDescriptor(scoreMemberAccessor, scoreDefinition);
+    }
+
+    private static MemberAccessor buildScoreMemberAccessor(DescriptorPolicy descriptorPolicy, Member member) {
+        return MemberAccessorFactory.buildMemberAccessor(
+                member,
+                FIELD_OR_GETTER_METHOD_WITH_SETTER,
+                PlanningScore.class,
+                descriptorPolicy.getDomainAccessType(),
+                descriptorPolicy.getGeneratedMemberAccessorMap());
+    }
+
+    private static Class<? extends Score<?>> extractScoreType(MemberAccessor scoreMemberAccessor, Class<?> solutionClass) {
```

Comment:
If these methods were not static and the class was generified as suggested, a lot of `? extends Score<?>` would go away for code that is easier to read.